### ISSUE TITLE
JSX: Build does not include dependencies of a changed module in bundle

### ISSUE THREAD

**Issue by nelsonlaquet** [jackfranklin/jspm-dev-builder#12]
I have the following gulpfile:

```
var gulp = require("gulp"),
    DevBuilder = require('jspm-dev-builder'),
    jspm = require("jspm"),
    path = require("path");

gulp.task("watch", function() {
    var appDevBuilder = new DevBuilder({
        jspm: jspm,
        expression: "scripts/bootstrap.js",
        outLoc: "public/bundle.js"
    });

    appDevBuilder.build();

    gulp.watch("public/scripts/**/*.{js,jsx}").on("change", function(file) {
        var changedPath = path.relative(path.join(__dirname, "public"), file.path);
        if (changedPath.endsWith(".jsx"))
            changedPath += "!";

        appDevBuilder.build(changedPath);
    });
});
```

Bootstrap is simply: `import "./application.jsx!"`

Application.jsx imports react, react-dom, lodash, and component.jsx!. Component.jsx simply imports react and exports a component.

Here's what happens:
- The first build works just fine. I get an entire bundle of bootstrap, application, component, react, react-dom, etc. 
- If I edit component.jsx OR bootstrap.js, it rebuilds correctly. 
- If I edit application.jsx - the rendered bundle does NOT include anything BUT application.jsx and bootstrap.js. React, react-dom, component, etc, are all no longer included.

Basically, it seems that if I edit a module that is the first in the hierarchy to import another module, that imported module is no longer included in any further bundles. This only seems to happen with jsx files, not with regular js files. It also appears that the extra runtime and polyfills are also ignored.

I'm using the following JSPM modules:

```
      "jsx": "npm:jspm-loader-jsx@^0.0.7",
      "lodash": "npm:lodash@^3.10.1",
      "react": "npm:react@^0.14.3",
      "react-dom": "npm:react-dom@^0.14.3"
      "babel": "npm:babel-core@^5.8.24",
      "babel-runtime": "npm:babel-runtime@^5.8.24",
      "core-js": "npm:core-js@^1.1.4"
```

And the following NPM modules:

```
    "jspm": "^0.16.15",
    "jspm-dev-builder": "^0.3.2"
```

I'm asking here because I wonder if anyone else has gotten around this issue with this tool. I believe this is: A) me using DevBuilder incorrectly, B) a bug in JSPM, or C) a bug in jspm-loader-jsx.

Honestly, this tool legitimizes the use of JSPM as a package loader for us stuck on HTTP/1 (and should probably be built into JSPM itself), and it'll be very sad if I can't get around this issue.

**Comment by jackfranklin** [comment]
This is really interesting and very odd. Can I ask firstly why you're not just letting Babel compile JSX for you? I've not found a need for an additional loader for JSX files.

Other than that I'd need to have the project to work with - are you able to recreate and put online?

Also, which version of jspm are you on?

**Comment by nelsonlaquet** [comment]
Here's a minimal reproduction of the bug:
https://github.com/nelsonlaquet/jspm-builder-jsx-bug

I've included the result of the first build, a build after editing application.jsx and a build after editing component.jsx. Here's the bugged one:
https://github.com/nelsonlaquet/jspm-builder-jsx-bug/blob/master/public/build/bundle-after-editing-app.js

I'm using jspm 0.16.15

As far as the loader, forgive me for being ignorant about jspm, but I was unable to get .jsx files to load without the loader - it would refuse to accept .jsx files, as it would append ".js" during file resolution. I did a brief search around and the JSX loader seemed to work. Though I'm noticing now that it's been depreciated, as the Facebook react transformer which it relies on has been as well.

My jsx files need to have jsx extensions, because my editor won't recognize the syntax without it. Is there a way to tell SystemJS to feed all js and jsx files though babel?

Thanks!

EDIT: I just noticed that in this repo, I'm using a different jsx loader than the one I had when the issue was started (this time I just typed in "jspm install jsx"), but the issue is the exact same regardless.
